lib/puppet/provider/foreman_resource/rest_v3.rb

```diff
@@ -99,14 +99,16 @@ def success?(response)
   end
 
   def error_message(response)
+    fqdn = URI::parse(resource[:base_url]).host
+
     explanations = {
-      '400' => 'Something is wrong with the data sent to Foreman server',
-      '401' => 'Often this is caused by invalid Oauth credentials',
-      '404' => 'The requested resource was not found',
-      '500' => 'Check /var/log/foreman/production.log on Foreman server for detailed information',
-      '502' => 'The webserver received an invalid response from the backend service. Was Foreman unable to handle the request?',
-      '503' => 'The webserver was unable to reach the backend service. Is foreman.service running?',
-      '504' => 'The webserver timed out waiting for a response from the backend service. Is Foreman under unusually heavy load?'
+      '400' => "Something is wrong with the data sent to Foreman at #{fqdn}",
+      '401' => "Often this is caused by invalid Oauth credentials sent to Foreman at #{fqdn}",
+      '404' => "The requested resource was not found in Foreman at #{fqdn}",
+      '500' => "Check /var/log/foreman/production.log on #{fqdn} for detailed information",
```

Comment:
```suggestion
      '500' => "Check Foreman logs at /var/log/foreman/production.log on #{fqdn} for detailed information",
```

Considering making this change as well

Reply:
I would keep it short. IMHO the path tells you enough what's there.

Reply:
OK, I basically agree. While this would be more consistent with the overall pattern, it doesn't really inform the user of anything new and there doesn't justify the added length.